pkg/detector/library/comparer/compare_test.go

```diff
@@ -0,0 +1,77 @@
+package comparer_test
+
+import (
+	"testing"
+
+	"github.com/aquasecurity/trivy/pkg/detector/library/comparer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersionComparer_MatchVersion(t *testing.T) {
```

Comment:
These tests are good but I think we should rather test the higher level abstraction of `IsVulnerable` as it's the one that is called from various call-sites and in turn calls `MatchVersion`. What do you think?

Reply:
I made `MatchVersion` unexported and added TestVersionComparer_IsVulnerable.
https://github.com/aquasecurity/trivy/commit/6361efc70da32126be62c841fc07bfbd053c2148